### ISSUE TITLE
Multiple tours

### ISSUE THREAD

**Issue by AlexanderVielhauer** [benmarch/angular-ui-tour#28]
HTML: http://pastebin.com/trp5pbMN
JS: `uiTourService.getTourByName("test").start()`

Expected: Both steps executed
Happens: Only first step is shown, then I can only select Pause or End

Works as expected if I remove the names.

**Comment by benmarch** [comment]
Hey @AlexanderVielhauer, for clarification, is your expectation that the two steps will display at the same time, or that they will display consecutively? From you code I would expect only one of the tours to display at any given time, and that you can only pause or end because each tour only contains one step. If you are intending to show multiple tour steps simultaneously, my recommendation is to do something similar to what you have now, except give each tour a unique name and then call start() on each of them; however, I'm sure there will be some quirks by implementing this, and I will not be able to support that functionality at this time. If you are looking to have multiple tours that are handled entirely separately, give each tour a unique name, and nest steps within them. Although your syntax will technically work, it is recommended that your tour steps are children of the tour element.

Thanks,
Ben

**Comment by AlexanderVielhauer** [comment]
What I wanted to do is to assign both tour steps to the same tour which is named `test` and then have it start with tour step 1 and proceed with tour step 2. So seems like I have to limit my usage of `ui-tour="test"` to one tour step and nest the consecutive steps into that? Like this: http://pastebin.com/yM5BKA1p

Thanks for your reply.

**Comment by benmarch** [comment]
Ah ok I see what you are trying to do, you need to wrap both tour steps in a single tour:

```html
<div ui-tour="test">
  <div tour-step .... ></div>
  <div tour-step .... ></div>
</div>
```

Let me know if that helps.

**Comment by AlexanderVielhauer** [comment]
Since the tour-steps need to be children of the respective `ui-tour` element, this way is not feasible for me because I want to tour elements in the same tour I cannot group like that. For now I just have one tour that covers everything and allow the user to jump to sub-tours in that one big tour. This works fine for me, so don't worry. :)

However it'd be cool to be free to group tour-steps regardless of dom structure somewhen in the future.

**Comment by benmarch** [comment]
Yeah sorry about that, I intentionally designed it this way originally and haven't thought much about it since. The intention was to keep it very declarative in the markup so that it was clear how and where the steps were defined (compared to Bootstrap Tour). I suppose at some point I could move the tour declaration out of the markup and allow the steps to reference a specific tour by name. That way the steps are still declarative, but I would have to come up with an intuitive way to create tours. I'll mull this over and see if there is a way to fit it in with the current design.

**Comment by benmarch** [comment]
Hey @AlexanderVielhauer, I have added a new experimental feature that will allow you to create what I am calling a "detached" tour (a tour that is not tied to the DOM). You can create tour steps exactly as you do now, except you would add a `tour-step-belongs-to="<tourName>"` attribute that would scope it accordingly. The feature is currently in the 28-detached-tour branch: https://github.com/benmarch/angular-ui-tour/tree/28-detached-tour

General usage would look like this:

```js
module.run(['uiTourService', function (uiTourService) {
  uiTourService.createDetachedTour('myDetachedTour', {backdrop: true});
}]);
```

```html
<!--anywhere in your markup-->
<div tour-step tour-step-belongs-to="myDetachedTour" tour-step-title="Detached Step" tour-step-content="This step will only be a part of the myDetachedTour tour" tour-step-order="0">
    ...
</div>
```
The second argument of `createDetachedTour()` is the configuration that is equivalent to setting attributes on a uiTour directive; they will override global tour options, and lifecycle hook functions must return promises for now. The steps for which you specify "belongsTo" will be ignored by other tours so they won't display multiple times.

Would you mind installing the 28-detached-tour version and testing this out on your end? Please let me know if this feature in its basic state is beneficial to you.

Thanks,
Ben

**Comment by AlexanderVielhauer** [comment]
Hi @benmarch, this sounds awesome! I am quite happy with my current solution (one big tour). The problem with multiple tours is that it needs all the triggers on the UI for the different tours, which I figured out would make my UI pretty loaded. However, I might test it out once I have some time to; but better not rely on that. :P

**Comment by benmarch** [comment]
Sounds good, released 0.5.0. Let me know if that works for you when you have chance.
